conda_pypi/post_command/install.py

```diff
@@ -0,0 +1,124 @@
+from __future__ import annotations
+
+from logging import getLogger
+from typing import TYPE_CHECKING
+
+from conda.base.context import context
+from conda.reporters import get_spinner
+from conda.exceptions import CondaVerificationError, CondaFileIOError
+
+from conda_pypi.main import run_pip_install, compute_record_sum, PyPIDistribution
+from conda_pypi.python_paths import get_env_site_packages
+
+if TYPE_CHECKING:
+    from typing import Iterable, Literal
+
+log = getLogger(f"conda.{__name__}")
+
+
+def _prepare_pypi_transaction(lines: Iterable[str]) -> dict[str, dict[str, str]]:
+    pkgs = {}
+    for line in lines:
+        dist = PyPIDistribution.from_lockfile_line(line)
+        pkgs[(dist.name, dist.version)] = {
+            "url": dist.find_wheel_url(),
+            "hashes": dist.record_checksums,
+        }
+    return pkgs
+
+
+def _verify_pypi_transaction(
+    prefix: str,
+    pkgs: dict[str, dict[str, str]],
+    on_error: Literal["ignore", "warn", "error"] = "warn",
+):
+    site_packages = get_env_site_packages(prefix)
+    errors = []
+    dist_infos = [path for path in site_packages.glob("*.dist-info") if path.is_dir()]
+    for (name, version), pkg in pkgs.items():
+        norm_name = name.lower().replace("-", "_").replace(".", "_")
+        dist_info = next(
+            (
+                d
+                for d in dist_infos
+                if d.stem.rsplit("-", 1) in ([name, version], [norm_name, version])
+            ),
+            None,
+        )
+        if not dist_info:
+            errors.append(f"Could not find installation for {name}=={version}")
+            continue
+
+        expected_hashes = pkg.get("hashes")
+        if expected_hashes:
+            found_hashes = compute_record_sum(dist_info / "RECORD", expected_hashes.keys())
+            log.info("Verifying %s==%s with %s", name, version, ", ".join(expected_hashes))
+            for algo, expected_hash in expected_hashes.items():
+                found_hash = found_hashes.get(algo)
+                if found_hash and expected_hash != found_hash:
+                    msg = (
+                        "%s checksum for %s==%s didn't match! Expected=%s, found=%s",
+                        algo,
+                        name,
+                        version,
+                        expected_hash,
+                        found_hash,
+                    )
+                    if on_error == "warn":
+                        log.warning(*msg)
+                    elif on_error == "error":
+                        errors.append(msg[0] % msg[1:])
+                    else:
+                        log.debug(*msg)
+    if errors:
+        errors = "\n- ".join(errors)
+        raise CondaVerificationError(f"PyPI packages checksum verification failed:\n- {errors}")
+
+
+def post_command(command: str):
```

Comment:
I would love to chat about this post_command sometime.  I want to understand better how these post_commands work.  My current mental model tells me that this is hooking into the `conda install` command.  If that is so then how do we assure we are invoking the post command?  Would calling it programmatically also invoke it?